src/Avalonia.Visuals/Media/TransformGroup.cs

```diff
@@ -17,6 +18,29 @@ public class TransformGroup : Transform
         public TransformGroup()
         {
             Children = new Transforms();
+            Children.CollectionChanged += (_, e) =>
```

Comment:
You can use [`ForEachItem`](https://github.com/AvaloniaUI/Avalonia/blob/d7036e0afe1b674401756b8966124c24b6620d1e/src/Avalonia.Base/Collections/AvaloniaListExtensions.cs#L41) to do this a little more succinctly I think. Also make sure that `Transforms` sets its `ResetBehavior` to `Remove`.